core/templates/dev/head/pages/exploration_player/AudioBarDirective.js

```diff
@@ -188,6 +180,17 @@ oppia.directive('audioBar', [
             }
           };
 
+          $scope.track = {             
+            progress: function(progress) {
+              if (angular.isDefined(progress)) {
```

Comment:
Some documentation here about what 'progress' is would be nice. A line like `progress: function(progress)` treats the term 'progress' as both a function and a variable and from reading this I have no idea what to pass in to the function. Perhaps rename the variable to progressPercentage if that's what it is?

In any case this still needs documentation, since one thing I'm confused about as a reader is why `progress` might not be defined.

Reply:
In this case ``ng-model`` is bind to a getter-setter function i.e. the function returns a representation of the model when called with zero arguments, and sets the internal state of a model when called with an argument. In case the user is not dragging the slider to move back and forth through the audio, the current progress of the track is automatically returned but if the slider is dragged to a position, the progress at that point will be passed as an argument to this function and then the track progress will be updated using that.

Reply:
I will add a comment for that!

Reply:
OK. The thing I don't get is how this changes the position of the "track head" in the audio file -- i.e., what is actually being played back. It just seems to be a UI thing, AFAICT. Could you elaborate on that?

Reply:
I defined a new function ``setProgress`` (https://github.com/bansalnitish/oppia/blob/1ce04e26da5146e63488ef57170861dd141ceaa2/core/templates/dev/head/services/AudioPlayerService.js#L113) to change the progress property of the audio being played. Here if the progress is passed as argument, the function is called to change the progress of audio file just as the ``rewind`` function (https://github.com/bansalnitish/oppia/blob/1ce04e26da5146e63488ef57170861dd141ceaa2/core/templates/dev/head/services/AudioPlayerService.js#L80) does.

Reply:
Got it. Thanks!